child_process/grpc_child_process_manager.go

```diff
@@ -0,0 +1,83 @@
+package childprocess
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"google.golang.org/grpc/status"
+)
+
+type ProcessClient interface {
+	Ping(nowait bool) error
+	Stop() error
+}
+
+type GRPCChildProcessManager struct {
+	processClient     ProcessClient
+	processBinaryPath string
+}
+
+func NewGRPCChildProcessManager(processClient ProcessClient, processBinaryPath string) GRPCChildProcessManager {
+	return GRPCChildProcessManager{
+		processClient:     processClient,
+		processBinaryPath: processBinaryPath,
+	}
+}
+
+func (g GRPCChildProcessManager) StartProcess() (StartupErrorCode, error) {
+	errChan := make(chan error)
+	go func() {
+		// #nosec G204 -- arg values are known before even running the program
+		_, err := exec.Command(g.processBinaryPath).Output()
+		errChan <- err
+	}()
+
+	pingChan := make(chan error)
+	// Start another goroutine where we ping the WaitForReady option, so that server has time to start up before we run
+	// the acctual command.
+	go func() {
+		err := g.processClient.Ping(false)
+		pingChan <- err
+	}()
+
+	select {
+	case err := <-errChan:
```

Comment:
Here you might receive `error` interface pointer with `nil` value. Is this part working ok?

Reply:
Can you explain? `errChan` is an `error` channel.

Reply:
Even in case 1st goroutine executes command and no error happens there, here `nil` value will be received and will be reported outside as failure to start binary

Reply:
Ah. In this case we always expect the started binary to outlive this function. If exec returns before the ping(second case in select), it means that something wrong has happened.

Reply:
Added separate error message for this case.